proposals/VK_EXT_external_memory_metal.adoc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021-2024 The Khronos Group Inc.
+//
+// SPDX-License-Identifier: CC-BY-4.0
+
+= VK_EXT_external_memory_metal
+:toc: left
+:refpage: https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/
+:sectnums:
+
+This extension allows memory sharing through imports from or export to Metal resource handles.
+
+== Problem Statement
+
+An application may wish to reference device memory in multiple Vulkan device instances, in multiple processes, and/or in Metal API.
+This extension enables an application to export non-Vulkan handles from Vulkan memory objects such that the underlying resources can
+be referenced outside the scope of the Vulkan device instance that created them.
+
+While link:{refpage}VK_EXT_metal_objects.html[VK_EXT_metal_objects] provides a way to expose underlying Metal resources,
+when importing an image from an `id<MTLTexture>` handle, said images and their backing memory will be imported at
+link:{refpage}VkImage.html[VkImage] creation.
+According to point link:https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#resources-association[12.8 Memory Resource Association]
+resources are created as _virtual allocations_ with no backing memory. This leads to having to treat imported images through
+link:{refpage}VK_EXT_metal_objects.html[VK_EXT_metal_objects] differently.
+
+This extension aims to import both resources through link:{refpage}VkDeviceMemory.html[VkDeviceMemory] so that later can be bound
+to a existing link:{refpage}VkImage.html[VkImage] or link:{refpage}VkBuffer.html[VkBuffer] respectively.
+
+== Proposal
+
+This extension aims to standardize memory imports and exports of Vulkan resources using
+link:{refpage}VK_KHR_external_memory.html[VK_KHR_external_memory] as baseline to conform to what other platforms have done with
+extensions like link:{refpage}VK_KHR_external_memory_win32.html[VK_KHR_external_memory_win32] and
+link:{refpage}VK_ANDROID_external_memory_android_hardware_buffer.html[VK_ANDROID_external_memory_android_hardware_buffer] to name a few.
+
+This means that external Metal resources will be imported through link:{refpage}VkDeviceMemory.html[VkDeviceMemory] to later be bound.
+
+3 new bit values will be added to link:{refpage}VkExternalMemoryHandleTypeFlagBits.html[VkExternalMemoryHandleTypeFlagBits]:
+
+ - `VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLBUFFER_BIT_EXT` to accommodate for Metal buffers (MTLBuffer)
+ - `VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT` to accommodate for Metal textures (MTLTexture)
+ - `VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLHEAP_BIT_EXT` to accommodate for Metal textures (MTLHeap)
```

Comment:
typo: 'Metal textures' instead of heaps.

Reply:
Done